### PR TITLE
*: move initial of membuff for temporary table to when table create

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Create membuff when creating the first local temporary table to increase readability and prevent potential creation overhead of membuff.

### What is changed and how it works?

Move initial of membuff for temporary table to when table create

### Check List 

Tests 

- Unit test

Side effects

- N/A

Documentation

- N/A

### Release note 

- No release note

### CODE REVIEW

File: executor/ddl.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/tikv/client-go/v2/tikv"
+
```
Comment (djshow832) [inline executor/ddl.go:22]: Remove the empty line and move the library down.

**Review by djshow832 — COMMENTED** [review]
Rest LGTM

**Review by tiancaiamao — APPROVED** [review]

**Review by djshow832 — APPROVED** [review]

**Pull request merged**